components/core/blocks/Product/SizeGuide.vue

```diff
@@ -16,20 +16,12 @@ import SizeGuideContent from 'theme/components/theme/blocks/Static/Example'
 
 export default {
   name: 'SizeGuide',
-  computed: {
-    ...mapState({
-      activeElem: state => state.ui.authElem
-    })
-  },
   components: {
     Modal,
     SizeGuideContent
   },
-  methods: {
-    close (e) {
-      if (e) localStorage.removeItem('redirect')
-      this.$bus.$emit('modal-hide', 'modal-sizeguide')
-    }
+  mounted () {
+    this.$bus.$emit('modal-show', 'modal-sizeguide')
```

Comment:
that looks weird :P I understand that you used v-if to handle visibility of that component, but modal shouldn't show itself. There is a common way to show modals by using `this.$bus.$emit('modal-show', 'modal-name')` outside of modal component. You break it in that place.
Suggestion:
Maybe make refactor for modals like is in capybara?
https://github.com/DivanteLtd/vsf-capybara/blob/develop/store/ui/modals.ts#L4-L27
https://github.com/DivanteLtd/vsf-capybara/blob/develop/components/organisms/o-modal.vue
Then you can lazy load all modals. your call

Reply:
second suggestion it to get rid of this mounted call and show it in PDP component by:
```
    openSizeGuide () {
      this.showSizeGuide = true;
      this.$bus.$emit('modal-show', 'modal-sizeguide')
    },
```

Reply:
Agree

Reply:
Decided to revert it to the previous way, at least for now